index.js

```diff
@@ -84,14 +89,17 @@ function _walkSync(baseDir, options, _relativePath) {
 
   for (var i=0; i<sortedEntries.length; ++i) {
     var entry = sortedEntries[i];
+    if (ignoreMatcher && ignoreMatcher.match(entry.relativePath)) {
```

Comment:
Oh, I see... Prevent it from ever being added to `entries` instead, right?  Makes sense to me.